hack/golangci-lint.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+VERSION="1.59.1"
+
+rootdir=$(git rev-parse --show-toplevel)
+if [ -z "${rootdir}" ]; then
+    echo "Failed to determine top level directory"
+    exit 1
+fi
+
+bindir="${rootdir}/bin"
+golangci_lint="${bindir}/golangci-lint"
+
+function get_tool {
+    mkdir -p "${bindir}"
+    echo "Downloading golangci-lint tool"
+    curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -d -b "${bindir}" "v${VERSION}"
+
+    if [ $? -ne 0 ]; then
+        echo "Install from script failed. Trying go install"
+        go install "github.com/golangci/golangci-lint/cmd/golangci-lint@v${VERSION}"
+        if [ $? -ne 0 ]; then
+            echo "Install of golangci-lint failed"
+            exit 1
+        else
+            golangci_lint=$(which golangci-lint)
```

Comment:
This location may not necessarily be where `go install` put it into... should be, but not necessarily.

Reply:
Updated to use `go env` to get the path